docs/fundamentals/k8s.md

```diff
@@ -0,0 +1,159 @@
+````
+
+To use Linkerd you can follow the [install the Linkerd CLI](https://linkerd.io/2.11/getting-started/).
+
+````{admonition} Caution
+:class: caution
+
+Many service meshes can perform retries themselves.
+Be careful about setting up service mesh level retries in combination with Jina, as it may lead to unwanted behaviour in combination with
+Jina's own {ref}`retry policy <flow-error-handling>`.
+
+Instead, you can disable Jina level retries by setting `Flow(retries=0)` in Python, or `retries: 0` in the Flow
+YAML `with` block.
+````
+
+
+(kubernetes-replicas)=
+## Scaling Executors: Replicas and shards
+
+Jina supports two ways of scaling:
+
+- **Replicas** can be used with any Executor type and are typically used for performance and availability.
+- **Shards** are used for partitioning data and should only be used with indexers since they store state.
+
+Check {ref}`here <scale-out>` for more information about these scaling mechanisms.
+
+For shards, Jina creates a separate Deployment in Kubernetes per Shard.
+Setting `f.add(..., shards=num_shards)` is sufficient to create a corresponding Kubernetes configuration.
+
+For replicas, Jina uses [Kubernetes native replica scaling](https://kubernetes.io/docs/tutorials/kubernetes-basics/scale/scale-intro/) and **relies on a service mesh** to load balance requests between replicas of the same Executor.
+Without a service mesh installed in your Kubernetes cluster, all traffic will be routed to the same replica.
+
+````{admonition} See Also
+:class: seealso
+
+The impossibility of load balancing between different replicas is a limitation of Kubernetes in combination with gRPC.
+If you want to learn more about this limitation, see [this](https://kubernetes.io/blog/2018/11/07/grpc-load-balancing-on-kubernetes-without-tears/) Kubernetes Blog post.
+````
+
+## Scaling the Gateway
+The Gateway is responsible for providing the API of the {ref}`Flow <flow>`.
+If you have a large Flow with many Clients and many replicated Executors, the Gateway can become the bottleneck.
+In this case you can also scale up the Gateway deployment to be backed by multiple Kubernetes Pods.
+This is done by the regular means of Kubernetes: Either increase the number of replicas in the  {ref}`generated yaml configuration files <kubernetes-deploy>` or [add replicas while running](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#scaling-a-deployment).
+To expose your Gateway replicas outside Kubernetes, you can add a load balancer as described {ref}`here <kubernetes-expose>`.
+
+````{admonition} Hint
+:class: hint
+You can use a custom Docker image for the Gateway deployment by setting the envrironment variable `JINA_GATEWAY_IMAGE` to the desired image before generating the configuration.
+````
+
+
+## Deploying On Jina AI Cloud
+
+Jina AI host its own Cloud based on Kubernetes cluster to manage your Flow:
```

Comment:
~Jina AI host~ Jina AI hosts? This sentence is a bit strange.

Check out Jina AI Cloud, the one-click solution to deploy and host Jina Flows, leveraging a cloud-native stack of Kubernetes, Prometheus and Grafana, without worrying about provisioning.

And then remove the Hint and update the JCloud page to also introduce Jina AI Cloud directly (?)